### DISCUSSION TITLE
AWS Secrets Manager integration

### DISCUSSION THREAD

**Discussion by mandzyu** [burningalchemist/sql_exporter#235] [Ideas]
Would you be open to a pull request enabling AWS SSM Secrets Manager as an optional field in the config?
My idea is that if a user has this set up on their AWS EC2 instance, they can provide the name of the secret ID in the config, and based on the contents of the secret, build the data source name to run the exporter. This way, they are not storing the username and password in a plaintext file.
I would be owning this implementation, and would submit the pull request.

**Comment by burningalchemist** [comment]
There is an open PR: https://github.com/burningalchemist/sql_exporter/pull/273. 
We can close this one as resolved.
